frontend/api/tinyUrl.ts

```diff
@@ -1,29 +1,27 @@
-const API_KEY = process.env.NEXT_PUBLIC_TINY_URL_API_KEY; 
+const API_KEY = process.env.NEXT_PUBLIC_TINY_URL_API_KEY;
 
-export async function createTinyUrl(url: string) {   
-    let body = {
-        url: url, 
-    }
+export async function createTinyUrl(url: string) {
+  let body = {
+    url: url,
+  };
 
-    return await fetch('https://api.tinyurl.com/create', {
-            method: 'POST', 
-            headers: {
-                accept: 'application/json',
-                authorization: `Bearer ${API_KEY}`, 
-                "content-type": `application/json`,
-        
-            },
-            body: JSON.stringify(body)
-            })
-            .then(response => {
-                if (response.status != 200) {
-                    throw `There was a problem with the fetch operation. Status Code: ${response.status}`;
-                }
-                return response.json()
-              })
-              .then(data => {
-                return data.data.tiny_url; 
-              })
-              .catch(error => console.error(error));      
-        }
-   
\ No newline at end of file
+  return await fetch("https://api.tinyurl.com/create", {
+    method: "POST",
+    headers: {
+      accept: "application/json",
+      authorization: `Bearer ${API_KEY}`,
+      "content-type": `application/json`,
+    },
+    body: JSON.stringify(body),
+  })
+    .then(response => {
+      if (response.status != 200) {
+        throw `There was a problem with the fetch operation. Status Code: ${response.status}`;
+      }
+      return response.json();
+    })
+    .then(data => {
+      return data.data.tiny_url;
+    })
+    .catch(error => console.error(error));
+}
```

Comment:
Met async-await is het niet de bedoeling rechtstreeks de Promise te gebruiken.

Dus bijvoorbeeld zo:

```
response = await fetch(...)
if (response.status != 200) { throw ... }

shortUrl = response.json().data.tiny_url
if (!shortUrl) { throw ... }

return shortUrl
```

En de caller wordt dan:

```
const longUrl = ...
try {
    const shortUrl = await createTinyUrl(longUrl);
    setSavedScenarioURL(shortUrl)
} catch e {
    console.error(e)
    setSavedScenarioURL(longUrl)
}
```